firmware/src/app.rs

```diff
@@ -94,9 +94,11 @@ impl<'a> App<'a> {
     fn process_request(&mut self, req: Request) {
         match req {
             Request::DAP1Command((report, n)) => {
-                let len =
-                    self.dap
-                        .process_command(&report[..n], &mut self.resp_buf, DAPVersion::V1);
+                let len = self.dap.process_command(
+                    &report[..n],
+                    &mut self.resp_buf[..64],
```

Comment:
This buffer size depends on the endpoint size, not on the DAP version. It's better to get it from the "hardware" layer in a reliable way, but I didn't implement it, because my assumption was "host shouldn't request more data than it can receive".

Reply:
I replaced the hardcoded packet sizes with constants. It's not perfect, and I have some ideas about how to refactor the usb packet stuff, but I didn't want this PR to get too big.